### PR TITLE
Publish minutes of 2022-01-20 meeting

### PR DESCRIPTION
**Pull request by Rob--W**
Generated from https://docs.google.com/document/d/1QkwhEMtMS67JBUkl_WVPZ4lRSKoWcQNlLJSf_GwSXg8/edit using the tool and process from #105.

During this meeting we discussed or mentioned #144, #145, #146, #120, #19, #115, #147, #72, #88, #134.

### CODE REVIEW

File: _minutes/2022-01-20-wecg.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+
+## Meeting notes
+
+[Issue 145](https://github.com/w3c/webextensions/issues/145): Proposals for improving WECG meetings
+
+ * [simeon] Introduced the topic in the last meeting, and elaborated in issue 145. Not that much to discuss today; encourage everyone to take a look and offer feedback.
+ * [tomislav] Timeboxing ([issue 144](https://github.com/w3c/webextensions/issues/144)) is a way to run a meeting. That suggestion isn't very controversial.
+ * [simeon] Idea of “champions” is just thinking aloud loud. Encourage everyone to take a look at the “Keeping meetings productive” section.
+ * [tomislav] Discussions that can be asynchronous/offline (i.e. in issues on Github) can better be discussed asynchronously.
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md:59]:
```suggestion
 * [tomislav] Discussions that can be asynchronous/offline (i.e. in issues on Github) can better be discussed asynchronously.
 * [jessie] The "champions" suggestion can be very effective for driving a topic or project to conclusion.
```

I tried to follow up with Jessie over email to make sure she's comfortable with this edit, but I haven't heard back. @xeenon, might you be able to check in?

File: _minutes/2022-01-20-wecg.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * [tomislav] Firefox does not intend to support the 5 minute limit that Chrome has.
+ * [bradley] We've tested the Safari preview, and cannot use non-blocking webRequest in its background script. What's the future of non-blocking webRequest across browsers?
+ * [jessie] Generally, try to avoid keeping the extension alive indefinitely. Would like to hear more about the use cases.
+ * [krysztof] There are existing examples on the issue tracker, e.g. [issue 72](https://github.com/w3c/webextensions/issues/72), [issue 88](https://github.com/w3c/webextensions/issues/88). The general problem with the declarative API is that advertising technology evolves rapidly and privacy/security extensions will not be able to keep up with declarative API alone.
+ * [simeon] Keeping service worker alive to observe network activity async is not ideal, but a currently acceptable middle-ground between constantly keeping a SW alive and totally removing webRequest.
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md]:
```suggestion
 * [simeon] Keeping service worker alive to observe asynchronously observe network activity is not ideal, but during the design process for MV3 we considered it acceptable middle-ground between constantly keeping the background context alive and totally removing webRequest. 
```

File: _minutes/2022-01-20-wecg.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ * [jessie] In Safari this API is not implemented.
+ * [simeon] Let's consider a process for aligning on extension APIs, e.g. using labels on Github.
+ * [rob] So far the decisions were captured in meeting notes, and in some cases repeated in comments on Github.
+
+[Issue 120](https://github.com/w3c/webextensions/issues/120): Manifest V3 vs native messaging [Ales Pospichal]
+
+ * [ales] Final decision / next steps on service worker limit? Native messaging app used by extension (smartcard). The app supports a content script via the background, and getting killed after 5 minutes interferes with the extension's functionality.
+ * [simeon] There are two relevant issues here in Chrome. Native messaging host terminated after 5 minutes, Native messaging host terminated too. Behavior not intentional and a high-prio issue.
+   * https://bugs.chromium.org/p/chromium/issues/detail?id=1152255
+   * https://bugs.chromium.org/p/chromium/issues/detail?id=1030305
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md]:
```suggestion
 * [simeon] There are two relevant issues here in Chrome: service worker terminated after 5 minutes (crbug.com/1152255) and native messaging host not being terminated on SW disconnect (crbug.com/1030305). Neither is intentional. Making it possible for extension SWs to live longer than 5 is one of our top priorities. 
```

File: _minutes/2022-01-20-wecg.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+ * [oliver] Is storage.session API going to be adopted by other browsers? Saw comments about structured cloning, can that be used to store non-JSON data?
+ * [tomislav] API is going to be the same. Structured cloning may be an option.
+ * [rob] We haven't started on implementation yet, but broadly it makes sense to have a consistent object serialization format across storage apis
+ * [jessie] (on behalf of Apple) Policy to not talk about unreleased APIs. The currently released MV3 API in Safari doesn't have storage.session.
+ * [oliver] cross-extension messaging: will runtime.sendMessage to message other extensions continue to be supported?
+ * [simeon] On Chrome this API should continue to work as expected. If it doesn't, that's a bug and we should fix it.
+ * [rob] On Firefox side, no plans to remove it either.
+ * [jessie] In Safari this API is not implemented.
+ * [simeon] Let's consider a process for aligning on extension APIs, e.g. using labels on Github.
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md]:
```suggestion
 * [simeon] Let's consider a process for more clearly communicating positions on extension APIs, e.g. using labels on Github.
```

File: _minutes/2022-01-20-wecg.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+ * [carlos] Let other browsers align on Firefox/Edge's way of specifying metadata such as minimum version.
+ * [simeon] Initial feedback from Chrome engineering is hesitation - we ignore unknown keys and could break existing extensions if we were to relocate properties. If the value of this feature is more clear we can reconsider if needed.
+ * [tomislav] We feel this is useful both for browser vendors and developers. For example, if we decide to support a feature similar to Chrome, but with a different shape, we either have to use another manifest key, or developers have to use separate manifests for different browsers.  In general, IMO top level keys are a good match for something that is agreed between browsers, and browser_specific_settings could be used for experimental things (until a final design is agreed between browsers) or for purposefully different things like min/max supported browser versions (which will forever be browser-specific).
+[Issue 147](https://github.com/w3c/webextensions/issues/147): Inconsistency: determine when non-persistent background pages get suspended()
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md:110]:
```suggestion

[Issue 147](https://github.com/w3c/webextensions/issues/147): Inconsistency: determine when non-persistent background pages get suspended()
```

At the moment everything form Issue 115 to Jessie's section is being very spaced out (each &lt;li> tag contains a &lt;p> tag). Inserting a new line above this line fixes the problem.

File: _minutes/2022-01-20-wecg.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+ * [carlos] Had previously raised this in the past. Created an issue because I believe this is something the group should do.
+ * [tomislav] The details are very implementation-specific, not sure if this is something that can be specified.
+ * [rob] Are we looking to clarify current behavior or attempt to standardize on a specific set of behaviors?
+ * [alexei] If everyone were to implement Mozilla's Limited Event page proposal ([issue 134](https://github.com/w3c/webextensions/issues/134)), then we would not have this inconsistency.
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md]:
```suggestion
 * [alexei] If everyone were to implement Mozilla's Limited Event page proposal ([issue 134](https://github.com/w3c/webextensions/issues/134)), then we would not have this inconsistency. Still have not heard a satisfactory reason why Google opposes this.
```

@ghostwords does this tweak look good to you?

**Review by zombie — APPROVED** [review]

File: _minutes/2022-01-20-wecg.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+ * [simeon] Objective is to minimize lifetime of pseudo-persistent background scripts. Concern with “event pages” is that developers may incorrectly assume that the current behavior of Chrome's MV2 event page is going to be supported. Currently working on getting our proposal into a publicly shareable state. Hoping to have shared in time for the next public session.
```
Comment (dotproto) [inline _minutes/2022-01-20-wecg.md]:
```suggestion
 * [simeon] Objective is to minimize lifetime of pseudo-persistent background scripts. Concern with “event pages” is that developers may incorrectly assume that the current behavior of Chrome's MV2 event page is going to be supported. Currently working on getting our proposal into a publicly shareable state. Hoping to have shared in time for the next public session. Comparing these proposals may help clarify our position better than I've been able to.
```

**Review by dotproto — APPROVED** [review]
I have a handful of clarifications. Most of them are for my own comments, but I tried to add a couple other points that stood out to me during the session.

File: _minutes/2022-01-20-wecg.md (hunk 6 above)
Comment (ghostwords) [reply _minutes/2022-01-20-wecg.md]: Yes, thank you!

I think the closest to an actual explanation thus far has been that [Google cares about low-end devices](https://github.com/w3c/webextensions/blob/main/_minutes/2021-12-09-wecg.md#:~:text=Especially%20a%20constraint%20for%20low%20end%20devices.), which, given that [SWs will probably use more not fewer system resources](https://www.eff.org/deeplinks/2021/12/googles-manifest-v3-still-hurts-privacy-security-innovation#:~:text=Google%E2%80%99s%20performance%20claims) overall, does not make any sense.

**Review by mukul-p — APPROVED** [review]

**Comment by Rob--W** [comment]
I have included all suggested edits in the source doc and updated the PR here. I also noticed that the year of the header was off, so I have prepended a commit to fix up the date in the previous meeting notes.

The diff is https://github.com/w3c/webextensions/compare/09a5b817b87e5154b6f9068a9238a9747ed8ca3d..0c60556493bba928b044ab022abf6d5633be6990

**Comment by Rob--W** [comment]
Since the meeting is tomorrow and the notes have been approved, with the review feedback included without changes, I'm going to merge this now.

**Pull request merged**